apps/people-app/microapp/src/pages/VehicleManagementPage.tsx

```diff
@@ -0,0 +1,180 @@
+      (loading) => setIsLoading(loading)
+    );
+  };
+
+  useEffect(() => {
+    fetchVehicles();
+  }, []);
+
+  const handleDeleteRequest = (id: number) => {
+    setVehiclePendingRemoval(id);
+    setShowDeleteModal(true);
+    setSelected(undefined);
+  };
+
+  const handleDeleteConfirm = async () => {
+    setShowDeleteModal(false);
+    setVehiclePendingRemoval(undefined);
+    fetchVehicles();
+  };
+
+  const handleDeleteCancel = () => {
+    setShowDeleteModal(false);
+    setVehiclePendingRemoval(undefined);
+  };
+
+  return (
+    <PageTransitionWrapper type="secondary">
+      <div className="h-screen">
+        <section className="px-1 pt-8">
+          <BackButton />
+        </section>
+        <div className="flex flex-col-reverse px-4 mt-5">
+          {isLoading ? (
+            <div className="grid place-items-center size-full border-b border-[#E5E5E5] pb-12">
+              <CircularProgress size={29} sx={{ color: "#ff7300" }} />
+            </div>
+          ) : (
+            <>
+              {rows.length > 0 ? (
+                <VehicleGroup
+                  selected={selected}
+                  onChange={(index: number | undefined) => {
+                    setSelected(index);
+                  }}
+                >
+                  {rows.map((row, index) => (
+                    <VehicleRow
+                      key={row.id}
+                      index={index + 1}
+                      {...row}
+                      onDelete={() => {
+                        handleDeleteRequest(row.id);
+                      }}
+                    />
+                  ))}
+                </VehicleGroup>
+              ) : (
+                <EmptyState />
+              )}
+            </>
+          )}
+        </div>
+      </div>
+      <section className="pr-1 mx-5 fixed bottom-20 right-1.5">
+        <AddButton onClick={() => setShowBottomSheet(true)} />
+      </section>
+      <AnimatePresence>
+        {showBottomSheet && (
+          <AddVehicleSheet
+            onClose={() => setShowBottomSheet(false)}
+            onSubmit={fetchVehicles}
+          />
+        )}
+        {showDeleteModal && vehiclePendingRemoval && (
+          <DeleteConfirmationModal
+            id={vehiclePendingRemoval}
+            onConfirm={handleDeleteConfirm}
+            onCancel={handleDeleteCancel}
+          />
+        )}
+      </AnimatePresence>
+    </PageTransitionWrapper>
+  );
+}
+
+export default VehicleManagementPage;
+
+interface AddButtonProps {
+  onClick: () => void;
+}
+
+function AddButton({ onClick }: AddButtonProps) {
+  return (
+    <IconButton
+      className="size-[40px] grid place-items-center rounded-full transition-colors"
+      onClick={onClick}
+      style={{
+        backgroundColor: "#ff7300",
+      }}
+    >
+      <AddSharp className="text-white" style={{ fontSize: 30.5 }} />
+    </IconButton>
+  );
+}
+
+export function BackButton() {
+  return (
+    <IconButton component={Link} to="/">
```

Comment:
To `"/"` will take the user to the root route. Not sure what the use case here is, but if you want to take the user to the previous screen in the stack instead of the home page, consider using `useNavigate` from `react-router-dom`

E.g.,
```typescript
import { useNavigate } from "react-router-dom"; // import
.
.
.
export function BackButton() {
  let navigate = useNavigate();
  const handleBackButtonClick = () => navigate(-1);
  return (
    <IconButton onClick={handleBackButtonClick}>
.
.
.
```